server/src/shared/serialize.interceptor.ts

```diff
@@ -0,0 +1,25 @@
+import { CallHandler, ExecutionContext, NestInterceptor, UseInterceptors } from '@nestjs/common';
+import { Observable, map } from 'rxjs';
+import { plainToInstance } from 'class-transformer';
+
+export const MapTo = (dto: any) => {
+  return UseInterceptors(new SerializeInterceptor(dto));
+};
+
+export class SerializeInterceptor implements NestInterceptor {
```

Comment:
Is an interceptor similar to a middleware? (I should go to the link and read up on `class-transformer` :D)

Reply:
In `class-transformer` [documentation](https://github.com/typestack/class-transformer#nodejs), it looks like we may also need `reflect-metadata` and don't see it as a dependency.

Reply:
It's satisfied by the nestjs/common peer dependencies, which uses it as well.

Reply:
> Is an interceptor similar to a middleware? 

In NestJs an interceptor is middle-ware. In this specific case, we're chaining a `class-transformer` operation on the outgoing handler response. It's applied to any handler (controller method) using the `UseInterceptors(new SerializeInterceptor(dto))` decorator, but in our case, I just wrapped it in a shorter decorator `MapTo`.